python-backend/app/api/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -41,16 +43,44 @@ def json(self):
             'issue_date':
             self.issue_date.isoformat() if self.issue_date else None,
             'authorization_end_date':
-            self.authorization_end_date.isoformat() if self.authorization_end_date else None
+            self.authorization_end_date.isoformat() if self.authorization_end_date else None,
+            'related_documents': [x.json() for x in self.documents]
         }
 
+    @classmethod
+    def create(cls,
+               permit,
+               received_date,
+               issue_date,
+               authorization_end_date,
+               user_kwargs,
+               permit_amendment_type_code='AMD',
+               permit_amendment_status_code='ACT',
+               save=False):
+        new = cls(
+            permit_id=permit.permit_id,
+            received_date=received_date,
+            issue_date=issue_date,
+            authorization_end_date=authorization_end_date,
+            permit_amendment_type_code=permit_amendment_type_code,
+            permit_amendment_status_code=permit_amendment_status_code,
+            **user_kwargs)
+        permit.permit_amendments.append(new)
+        if save:
+            db.session.commit()
+        return new
+
     @classmethod
     def find_by_permit_amendment_id(cls, _id):
-        return cls.query.filter_by(permit_amendment_id=_id).first()
+        return cls.query.filter_by(permit_amendment_id=_id).filter_by(deleted_ind=False).first()
+
+    @classmethod
+    def find_by_permit_amendment_guid(cls, _guid):
+        return cls.query.filter_by(permit_amendment_guid=_guid).filter_by(deleted_ind=False).first()
 
     @classmethod
     def find_by_permit_id(cls, _id):
-        return cls.query.filter_by(permit_id=_id)
+        return cls.query.filter_by(permit_id=_id).filter_by(deleted_ind=False).all()
```

Comment:
It seems like it could be helpful to add `.filter_by(deleted_ind=False)` to a base query for this class. And maybe do something similar for `active_ind` on some other classes?

Any idea if we could do something like this:
```python
def not_deleted(cls):
    return cls.query.filter_by(deleted_ind=False)

@classmethod
def find_by_permit_id(cls, _id):
    return cls.not_deleted().filter_by(permit_id=_id).all()
```
or
```python
not_deleted = PermitAmendment.query.filter_by(deleted_ind=False)

@classmethod
def find_by_permit_id(cls, _id):
    return not_deleted.filter_by(permit_id=_id).all()
```

None of this is tested -- just spit-balling ideas. Thoughts?

Reply:
I'm been looking to add it to the default behaviour of the db.query object. like how we have the tenant level control for mines. https://bcmines.atlassian.net/browse/MDS-1460, all queries automatically have the deleted filter unless you call a 'include_deleted()' method. The design principle is that deleted objects would never be returned by the api (sql forensics only).

Reply:
That would be my preference as well. I just wasn't sure if that sort of global change could be made in the microservice approach that we're supposed to be implementing